src/test/java/org/broadinstitute/hellbender/tools/dataflow/TestFewReadsWriter.java

```diff
@@ -0,0 +1,174 @@
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.channels.Channels;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Tests for FewReadsWriter.
+ */
+public class TestFewReadsWriter extends BaseTest {
+    private static final String LOCAL_INPUT = "src/test/resources/org/broadinstitute/hellbender/engine/reads_data_source_test2.bam";
+    private static final String CLOUD_INPUT = DATAFLOW_TEST_INPUTS + "org/broadinstitute/hellbender/engine/reads_data_source_test2.bam";
+
+    private SAMFileHeader header;
+
+    @Test
+    public void checkLocal() throws Exception {
+        File out = createTempFile("temp",".bam");
+        String outputFile = out.getPath();
+        testReadAndWrite(LOCAL_INPUT, outputFile, false);
+    }
+
+    @Test(groups = {"bucket"})
+    public void checkGCSInput() throws Exception {
+        File out = createTempFile("temp",".bam");
+        String outputFile = out.getPath();
+        testReadAndWrite(CLOUD_INPUT, outputFile, false);
+    }
+
+    // this leaves output files around, for now.
+    @Test(groups = {"bucket"})
+    public void checkGCSOutput() throws Exception {
+        File out = createTempFile("temp",".bam");
+        String tempName = out.getName();
+        String outputPath = DATAFLOW_TEST_STAGING + tempName;
+        testReadAndWrite(LOCAL_INPUT, outputPath, false);
+    }
+
+    // this leaves output files around, for now.
+    @Test(groups = {"cloud"})
+    public void checkCloud() throws Exception {
+        File out = createTempFile("temp",".bam");
+        String tempName = out.getName();
+        String outputPath = DATAFLOW_TEST_STAGING + tempName;
+        testReadAndWrite(CLOUD_INPUT, outputPath, true);
+    }
+
+    protected void testReadAndWrite(final String inputPath, final String outputPath, boolean remote) throws IOException {
+        String localInput = inputPath;
+        String localOutput = outputPath;
+        logger.info("outputPath="+outputPath);
+        final Pipeline pipeline = setupPipeline(remote);
+        if (BucketUtils.isCloudStorageUrl(inputPath)) {
+            File out = createTempFile("temp-input",".bam");
+            localInput = out.getPath();
+            localInput="temp-input.bam";
+            logger.info("Downloading the input to "+localInput+" (for comparison).");
+            downloadFromGCS(pipeline.getOptions(), inputPath, localInput);
+        }
+        PCollection<Read> input = ingestReadsAndGrabHeader(pipeline, inputPath);
+        FewReadsWriter.WriteToFile(pipeline, input, header, outputPath);
+        pipeline.run();
+        if (BucketUtils.isCloudStorageUrl(outputPath)) {
+            File out = createTempFile("temp-output",".bam");
+            localOutput = out.getPath();
+            localOutput="temp-output.bam";
```

Comment:
This also looks wrong (assigning to `localOutput` twice in a row) -- should be `out.getPath()`, right?